alertmanager-client/shell.nix

```diff
@@ -0,0 +1,35 @@
+{ nixpkgs ? import ../nix/pin.nix {}, compiler ? "default", doBenchmark ? false }:
+
+let
+
+  inherit (nixpkgs) pkgs;
+
+  f = { mkDerivation, alertmanager-openapi, base, http-client, lens
+      , lib, text, transformers, cabal-install
+      }:
+      mkDerivation {
+        pname = "alertmanager-client";
+        version = "0.1.0.0";
+        src = ./.;
+        libraryToolDepends = [ cabal-install ];
+        libraryHaskellDepends = [
+          alertmanager-openapi base http-client lens text transformers
+        ];
+        homepage = "https://github.com/SupercedeTech/alertmanager-client#readme";
+        license = lib.licenses.bsd3;
+      };
+
+  haskellPackages = (if compiler == "default"
+                       then pkgs.haskellPackages
+                     else pkgs.haskell.packages.${compiler}).extend(self: super: {
+                        alertmanager-openapi = haskellPackages.callPackage ../alertmanager-openapi {};
```

Comment:
Why is explicitly adding the package necessary here but not above?

Reply:
because many, many more packages are broken, I considered going from me not being able to build it at all to having it generate and build the new bindings good enough for this upgrade cycle.

Reply:
it's not available in the package set. no-one uploaded it to hackage.
Even if it was you want to override it to the current version in the tree rather then the one on hackage because it may contain changes.

Reply:
Sorry, my question wasn't clear - why isn't it necessary to add `alertmanager-openapi` to the package set when the compiler choice is `"default"`? It won't be in the Hackage set in either case.

Reply:
otherwise cabal can't find it, nix patches cabal somehow